### PR TITLE
encryption-at-rest: describe compatibility change with TiKV <= 4.0.8

### PR DESCRIPTION
**Pull request by Joyinqin**
### What is changed, added or deleted? (Required)

describe compatibility change with TiKV <= 4.0.8
### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/4351
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by Joyinqin** [comment]
/label needs-cherry-pick-4.0, 
size/medium, status/DNM, translation/from-docs

File: encryption-at-rest.md [hunk 1]
```diff
@@ -107,6 +107,19 @@ region = "us-west-2"
 
 在调试方面，可使用 `tikv-ctl` 命令查看加密元数据（例如使用的加密方法和数据密钥列表）。该操作可能会暴露密钥，因此不推荐在生产环境中使用。详情参阅 [TiKV Control](/tikv-control.md#打印加密元数据)。
 
+### 与早期 TiKV 版本的兼容性
+
+为了减少 TiKV 在管理加密元数据时由 I/O 操作和互斥体争用引发的开销，TiKV v4.0.9 对此进行了优化，并由 TiKV 配置文件中的 `security.encryption.enable-file-dictionary-log` 参数来控制优化操作。此配置参数仅在 TiKV v4.0.9 或更高版本中才生效。
```
Comment (TomShawn) [inline encryption-at-rest.md:112]:
```suggestion
为了减少 TiKV 在管理加密元数据时因 I/O 操作和互斥锁争用导致的开销，TiKV v4.0.9 对此进行了优化，由 TiKV 配置文件中的 `security.encryption.enable-file-dictionary-log` 参数来控制优化操作。此配置参数仅在 TiKV v4.0.9 或更高版本中生效。
```

File: encryption-at-rest.md [hunk 2]
```diff
@@ -107,6 +107,19 @@ region = "us-west-2"
...
+
+默认情况下启用加密功能时，TiKV v4.0.8 或更早期的版本无法识别加密元数据的数据格式。例如，假设你正在使用具有静态加密和默认的 `enable-file-dictionary-log` 配置的 TiKV v4.0.9 或更高版本时，如果将集群降级到 TiKV v4.0.8 或更早版本，则 TiKV 将无法启动，并且信息日志中会出现类似报错，如下所示：
```
Comment (TomShawn) [inline encryption-at-rest.md:114]:
```suggestion
默认情况下启用加密功能，此时 TiKV v4.0.8 或更早期的版本无法识别加密元数据的数据格式。例如，假设你正在使用的 TiKV v4.0.9 或更高版本开启了静态加密和默认开启了 `enable-file-dictionary-log` 配置，如果将集群降级到 TiKV v4.0.8 或更早版本，则 TiKV 将无法启动，并且信息日志中会类似如下的报错：
```

File: encryption-at-rest.md [hunk 3]
```diff
@@ -107,6 +107,19 @@ region = "us-west-2"
...
+
+```
+[2020/12/07 07:26:31.106 +08:00] [ERROR] [mod.rs:110] ["encryption: failed to load file dictionary."]
+[2020/12/07 07:26:33.598 +08:00] [FATAL] [lib.rs:483] ["called `Result::unwrap()` on an `Err` value: Other(\"[components/encryption/src/encrypted_file/header.rs:18]: unknown version 2\")"]
+```
+
+为了避免上述错误，你可以首先将 `security.encryption.enable-file-dictionary-log` 设置为 `false`，然后启动 TiKV v4.0.9 或更高版本。TiKV 成功启动后，加密元数据的数据格式将降级为 TiKV 早期版本可以识别的格式。此时，你可以将 TiKV 集群降级到较早的版本。
```
Comment (TomShawn) [inline encryption-at-rest.md:121]:
```suggestion
为了避免上面所示错误，你可以首先将 `security.encryption.enable-file-dictionary-log` 设置为 `false`，然后启动 TiKV v4.0.9 或更高版本。TiKV 成功启动后，加密元数据的数据格式将降级为 TiKV 早期版本可以识别的格式。此时，你可再将 TiKV 集群降级到较早的版本。
```

File: tikv-configuration-file.md [hunk 4]
```diff
@@ -1151,6 +1151,36 @@ raftdb 相关配置项。
 + 若开启该选项，日志中的用户数据会以 `?` 代替。
 + 默认值：`false`
 
+## security.encryption
+
+与[静态加密](/encryption-at-rest.md)(TDE)有关的配置项。
```
Comment (TomShawn) [inline tikv-configuration-file.md:1156]:
```suggestion
[静态加密](/encryption-at-rest.md) (TDE) 有关的配置项。
```

File: tikv-configuration-file.md [hunk 5]
```diff
@@ -1151,6 +1151,36 @@ raftdb 相关配置项。
...
+
+### `data-key-rotation-period`
+
++ 指定 TiKV 轮换数据密钥的频率。
++ 默认值：`7d`
+
+### enable-file-dictionary-log
+
++ 启用优化，以减少 TiKV 管理加密元数据时的 I/O 操作和互斥体争用。
++ 避免默认情况下启用此配置参数时可能出现的兼容性问题。具体可以参考[静态加密](/encryption-at-rest.md#与早期-tikv-版本的兼容性)。
```
Comment (TomShawn) [inline tikv-configuration-file.md:1173]:
```suggestion
+ 此配置参数默认启用，为避免可能出现的兼容性问题，请参考[静态加密 - 与早期 TiKV 版本的兼容性](/encryption-at-rest.md#与早期-tikv-版本的兼容性)。
```

File: tikv-configuration-file.md [hunk 6]
```diff
@@ -1151,6 +1151,36 @@ raftdb 相关配置项。
...
++ 默认值：`true`
+
+### master-key
+
++ 指定启用加密时的主密钥。想要了解如何配置主密钥，可以参考[静态加密](/encryption-at-rest.md#配置加密)。
```
Comment (TomShawn) [inline tikv-configuration-file.md:1178]:
```suggestion
+ 指定启用加密时的主密钥。若要了解如何配置主密钥，可以参考[静态加密 - 配置加密](/encryption-at-rest.md#配置加密)。
```

File: tikv-configuration-file.md [hunk 7]
```diff
@@ -1151,6 +1151,36 @@ raftdb 相关配置项。
...
+
+### previous-master-key
+
++ 指定轮换新主密钥时的旧主密钥。旧主密钥的配置格式与主密钥相同。想要了解如何配置主密钥，可以参考[静态加密](/encryption-at-rest.md#配置加密)。
```
Comment (TomShawn) [inline tikv-configuration-file.md:1182]:
```suggestion
+ 指定轮换新主密钥时的旧主密钥。旧主密钥的配置格式与主密钥相同。若要了解如何配置主密钥，可以参考[静态加密 - 配置加密](/encryption-at-rest.md#配置加密)。
```

File: encryption-at-rest.md [hunk 8]
```diff
@@ -107,6 +107,19 @@ region = "us-west-2"
 
 在调试方面，可使用 `tikv-ctl` 命令查看加密元数据（例如使用的加密方法和数据密钥列表）。该操作可能会暴露密钥，因此不推荐在生产环境中使用。详情参阅 [TiKV Control](/tikv-control.md#打印加密元数据)。
 
+### 与早期 TiKV 版本的兼容性
```
Comment (yiwu-arbug) [inline encryption-at-rest.md:110]:
```suggestion
### TiKV 版本间兼容性
```

File: encryption-at-rest.md (hunk 1 above)
Comment (yiwu-arbug) [inline encryption-at-rest.md:112]:
```suggestion
为了减少 TiKV 管理加密元数据造成的 I/O 操作和互斥锁竞争引发的开销，TiKV v4.0.9 对此进行了优化。此优化默认打开，并且可以通过 TiKV 配置文件中的 `security.encryption.enable-file-dictionary-log` 参数关闭。此配置参数仅在 TiKV v4.0.9 或更高版本中生效。
```

File: tikv-configuration-file.md [hunk 9]
```diff
@@ -1151,6 +1151,36 @@ raftdb 相关配置项。
...
++ 默认值：`"plaintext"`
+
+### `data-key-rotation-period`
+
++ 指定 TiKV 轮换数据密钥的频率。
++ 默认值：`7d`
+
+### enable-file-dictionary-log
+
++ 启用优化，以减少 TiKV 管理加密元数据时的 I/O 操作和互斥体争用。
```
Comment (yiwu-arbug) [inline tikv-configuration-file.md:1172]:
```suggestion
+ 启用优化，以减少 TiKV 管理加密元数据造成的 I/O 操作和互斥锁竞争。
```

File: encryption-at-rest.md (hunk 8 above)
Comment (yiwu-arbug) [reply encryption-at-rest.md:110]: 需要同时修改相应链接

File: encryption-at-rest.md [hunk 10]
```diff
@@ -47,9 +47,9 @@ data-encryption-method = aes128-ctr
 data-key-rotation-period = 7d
 ```
 
-`data-encryption-method` 的可选值为 `aes128-ctr`、`aes192-ctr`、`aes256-ctr` 和 `plaintext`。默认值为 `plaintext`，即默认不开启加密功能。`data-key-rotation-period` 指定 TiKV 轮换密钥的频率。可以为新 TiKV 群集或现有 TiKV 群集开启加密，但只有启用后写入的数据才保证被加密。要禁用加密，请在配置文件中删除 `data-encryption-method`，或将该参数值为 `plaintext`，然后重启 TiKV。
+`data-encryption-method` 的可选值为 `"aes128-ctr"`、`"aes192-ctr"`、`"aes256-ctr"` 和 `"plaintext"`。默认值为 `"plaintext"`，即默认不开启加密功能。`data-key-rotation-period` 指定 TiKV 轮换密钥的频率。可以为新 TiKV 群集或现有 TiKV 群集开启加密，但只有启用后写入的数据才保证被加密。要禁用加密，请在配置文件中删除 `data-encryption-method`，或将该参数值为 `"plaintext"`，然后重启 TiKV。
 
-如果启用了加密（即 `data-encryption-method` 的值不是 `plaintext`），则必须指定主密钥。要使用 AWS KMS 方式指定为主密钥，请在`[security.encryption]` 部分之后添加 `[security.encryption.master-key]` 部分：
+如果启用了加密（即 `data-encryption-method` 的值不是 `"plaintext"`），则必须指定主密钥。要使用 AWS KMS 方式指定为主密钥，请在`[security.encryption]` 部分之后添加 `[security.encryption.master-key]` 部分：
```
Comment (TomShawn) [inline encryption-at-rest.md:52]:
```suggestion
如果启用了加密（即 `data-encryption-method` 的值不是 `"plaintext"`），则必须指定主密钥。要使用 AWS KMS 方式指定为主密钥，请在 `[security.encryption]` 部分之后添加 `[security.encryption.master-key]` 部分：
```

File: encryption-at-rest.md [hunk 11]
```diff
@@ -107,6 +107,19 @@ region = "us-west-2"
 
 在调试方面，可使用 `tikv-ctl` 命令查看加密元数据（例如使用的加密方法和数据密钥列表）。该操作可能会暴露密钥，因此不推荐在生产环境中使用。详情参阅 [TiKV Control](/tikv-control.md#打印加密元数据)。
 
+### TiKV 版本间兼容性
```
Comment (TomShawn) [inline encryption-at-rest.md:110]: @yiwu-arbug The English title is `Compatibility with earlier TiKV versions`. Can we make the two versions literally as close as possible?

File: encryption-at-rest.md [hunk 12]
```diff
@@ -107,6 +107,19 @@ region = "us-west-2"
...
+
+为了减少 TiKV 管理加密元数据造成的 I/O 操作和互斥锁竞争引发的开销，TiKV v4.0.9 对此进行了优化。此优化可以通过 TiKV 配置文件中的 `security.encryption.enable-file-dictionary-log` 参数启用或关闭。此配置参数仅在 TiKV v4.0.9 或更高版本中生效。
```
Comment (TomShawn) [inline encryption-at-rest.md:112]:
```suggestion
为了减少 TiKV 管理加密元数据造成的 I/O 操作和互斥锁竞争以及由此引发的开销，TiKV v4.0.9 对此进行了优化。此优化可以通过 TiKV 配置文件中的 `security.encryption.enable-file-dictionary-log` 参数启用或关闭。此配置参数仅在 TiKV v4.0.9 或更高版本中生效。
```

File: encryption-at-rest.md [hunk 13]
```diff
@@ -107,6 +107,19 @@ region = "us-west-2"
...
+
+该配置项默认开启，此时 TiKV v4.0.8 或更早期的版本无法识别加密元数据的数据格式。例如，假设你正在使用的 TiKV v4.0.9 或更高版本开启了静态加密和默认开启了 `enable-file-dictionary-log` 配置，如果将集群降级到 TiKV v4.0.8 或更早版本，TiKV 将无法启动，并且信息日志中会有类似如下的报错：
```
Comment (TomShawn) [inline encryption-at-rest.md:114]:
```suggestion
该配置项默认开启，此时 TiKV v4.0.8 或更早期的版本无法识别加密元数据的数据格式。例如，假设你正在使用 TiKV v4.0.9 或更高版本，开启了静态加密和默认开启了 `enable-file-dictionary-log` 配置。如果将集群降级到 TiKV v4.0.8 或更早版本，TiKV 将无法启动，并且信息日志中会有类似如下的报错：
```

File: encryption-at-rest.md (hunk 11 above)
Comment (yiwu-arbug) [reply encryption-at-rest.md:110]: updated the English version as well.

File: encryption-at-rest.md [hunk 14]
```diff
@@ -107,18 +107,18 @@ region = "us-west-2"
 
 在调试方面，可使用 `tikv-ctl` 命令查看加密元数据（例如使用的加密方法和数据密钥列表）。该操作可能会暴露密钥，因此不推荐在生产环境中使用。详情参阅 [TiKV Control](/tikv-control.md#打印加密元数据)。
 
-### 与早期 TiKV 版本的兼容性
+### TiKV 版本间兼容性
 
-为了减少 TiKV 在管理加密元数据时由 I/O 操作和互斥体争用引发的开销，TiKV v4.0.9 对此进行了优化，并由 TiKV 配置文件中的 `security.encryption.enable-file-dictionary-log` 参数来控制优化操作。此配置参数仅在 TiKV v4.0.9 或更高版本中才生效。
+为了减少 TiKV 在管理加密元数据时因 I/O 操作和互斥锁争用导致的开销，TiKV v4.0.9 对此进行了优化，由 TiKV 配置文件中的 `security.encryption.enable-file-dictionary-log` 参数来控制优化操作。此配置参数仅在 TiKV v4.0.9 或更高版本中生效。
```
Comment (yiwu-arbug) [inline encryption-at-rest.md]:
```suggestion
为了减少 TiKV 管理加密元数据造成的额外 I/O 操作和互斥锁争用导致的开销，TiKV v4.0.9 对此进行了优化，由 TiKV 配置文件中的 `security.encryption.enable-file-dictionary-log` 参数来控制优化操作。此配置参数仅在 TiKV v4.0.9 或更高版本中生效。
```

File: tikv-configuration-file.md [hunk 15]
```diff
@@ -1169,17 +1169,17 @@ raftdb 相关配置项。
 
 ### enable-file-dictionary-log
 
-+ 启用优化，以减少 TiKV 管理加密元数据时的 I/O 操作和互斥体争用。
-+ 避免默认情况下启用此配置参数时可能出现的兼容性问题。具体可以参考[静态加密](/encryption-at-rest.md#与早期-tikv-版本的兼容性)。
++ 启用优化，以减少 TiKV 管理加密元数据时的 I/O 操作和互斥锁争用。
++ 此配置参数默认启用，为避免可能出现的兼容性问题，请参考[静态加密 - 与早期 TiKV 版本的兼容性](/encryption-at-rest.md#与早期-tikv-版本的兼容性)。
```
Comment (yiwu-arbug) [inline tikv-configuration-file.md]:
```suggestion
+ 此配置参数默认启用，为避免可能出现的兼容性问题，请参考[静态加密 - TiKV 版本间兼容性](/encryption-at-rest.md#tikv-版本间兼容性)。
```

File: encryption-at-rest.md (hunk 11 above)
Comment (TomShawn) [reply encryption-at-rest.md:110]: OK

**Comment by TomShawn** [comment]
LGTM

**Comment by yiwu-arbug** [comment]
LGTM

**Comment by yiwu-arbug** [comment]
The external link error is not related.

**Comment by TomShawn** [comment]
> The external link error is not related.

We can ignore that.

**Comment by yiwu-arbug** [comment]
> > The external link error is not related.
> 
> We can ignore that.

then, shall we merge, or is there anything that still need to resolve?

**Comment by TomShawn** [comment]
> > > The external link error is not related.
> > 
> > 
> > We can ignore that.
> 
> then, shall we merge, or is there anything that still need to resolve?

Wait until v4.0.9 is released. In most situations, user docs should go after codes.

**Comment by yiwu-arbug** [comment]
> Wait until v4.0.9 is released because this PR explicitly says "v4.0.9". Users must be very confused if they see an unreleased version in docs.

got it. make sense.

**Review by TomShawn — APPROVED** [review]

**Pull request merged**